src/ast/passes/codegen_llvm.cpp

```diff
@@ -3909,6 +3959,154 @@
   return callback;
 }
 
+Function *CodegenLLVM::createMapDeleteFilteredCallback(location loc)
+{
+  if (map_delete_filtered_cb_ != nullptr)
+    return map_delete_filtered_cb_;
+
+  /**
+   * The goal is to produce this code:
+   *
+   * struct ctx_t {
+   *   uint64_t offset;
+   *   uint64_t value;
```

Comment:
Perhaps good to call out somewhere in this function or the above that integers are always stored as
64bit values in maps. I see that mentioned in AssignMapStatement code but might be useful to point
it out again